cloud/cloud.py

```diff
@@ -1,18 +1,29 @@
+import libcloud
+import toml
+
 import cloud
 from cloud import registry as reg
 from cloud import Instance
 
 
-def connect(arg):
-
-  if isinstance(arg, str):
+def connect(arg=None):
+  if arg is None:
+    with open(path, "r") as cf:
+      config = toml.loads(cf)
+      provider = config.pop("provider").lower()
+      cloud.instance = reg.retrieve(provider, config=config)
+  elif isinstance(arg, str):
     cloud.instance = reg.retrieve(arg)
   elif isinstance(arg, Instance):
```

Comment:
when would this be used?

Reply:
If they have their own custom Instance object they want to use